enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -86,6 +83,22 @@ inventory.
 
 ### Implementation Details/Notes/Constraints
 
+#### Co-existing with SaaS
+
+The assisted service is currently deployed as a SaaS on cloud.redhat.com with
+a non-k8s REST API and a SQL database. The software implementing that needs to
+co-exist with the software that runs in a cluster and exposing a k8s-native API,
+while sharing most of the same features and behaviors.
```

Comment:
Saying "co-exist" makes it seem as though there would be some sort of technical conflict between the two implementations if the same cluster could see both, but I'm having trouble imaging what that would be. Is this really just trying to say "we want to reuse the existing service code both in the cloud and for on-prem use because <insert reason(s)>"? Based on line 98, I think so?

Reply:
The SaaS and in-cluster implementations would not necessarily co-exist from the perspective of a particular cluster. This section just talks about them co-existing in the world. :) We need to deliver very similar functionality in two very different ways. The implementation details of that pose a challenge.

Reply:
OK, that isn't clear from the current version of this section. Your comment is clearer, but doesn't give the motivation for choosing to wrap the existing software instead of writing something from scratch that would be k8s-native to begin with. I'm not suggesting that the choice is wrong, just that it isn't documented so that it's clear why this approach is recommended over the other and if there is a future in which a different choice might make more sense.

I think the background is that the existing implementation was not built with this use case in mind, and it is built in such a way that separating the agent management logic and database layer are non-trivial in the time we have to deliver this feature. Is that right?

Reply:
Thanks for the feedback. I'll re-work this section and see if the result is more clear.